docs/articles/documentation/test-api/intercepting-and-mocking-http-requests.md

```diff
@@ -0,0 +1,547 @@
+`response.body`    | [Buffer](https://nodejs.org/api/buffer.html) &#124; String | The response body. A [Buffer](https://nodejs.org/api/buffer.html) or string depending on the `stringifyResponseBody` option. Logged if the `logResponseBody` option is set to `true`.
+
+```js
+import { RequestLogger } from 'testcafe';
+
+const logger = RequestLogger('http://api.example.com');
+
+fixture `test`
+    .page('http://example.com');
+
+test
+    .requestHooks(logger)
+    ('test', async t => {
+        const logRecord = logger.requests[0];
+
+        console.log(logRecord.id);                  // UnqLnm189
+        console.log(logRecord.testRunId);           // IwQA12J12
+        console.log(logRecord.userAgent);           // Chrome 63.0.3239 / Windows 8.1.0.0
+        console.log(logRecord.request.url);         // http://api.example.com
+        console.log(logRecord.request.method);      // get
+        console.log(logRecord.response.statusCode); // 304
+    });
+```
+
+**Example**
+
+```js
+import { RequestLogger } from 'testcafe';
+
+const logger = RequestLogger('http://api.example.com');
+
+fixture `test`
+    .page('http://example.com');
+
+test
+    .requestHooks(logger)
+    ('test', async t => {
+        await t.expect(logger.contains(record => record.response.statusCode === 200)).ok();
+    });
+```
+
+## Mocking HTTP Responses
+
+Mocking is useful when the tested app uses a piece of infrastructure that is difficult to deploy during the test run. In this instance, you can intercept requests to this resource and mock the responses using TestCafe.
+
+To create a response mocker, use the `RequestMock` constructor.
+
+```js
+var mock = RequestMock()
+```
+
+Then call the `onRequestTo` and `respond` methods in a chain. The `onRequestTo` method specifies a request to intercept, while the `respond` method specifies the mocked response for this request. Repeat calling these methods to provide a mock for every request you need.
+
+```js
+var mock = RequestMock()
+    .onRequestTo(request1)
+    .respond(responseMock1)
+    .onRequestTo(request2)
+    .respond(responseMock2);
+```
+
+To enable the hook to mock the requests, [attach it to a test or fixture](#attaching-hooks-to-tests-and-fixtures).
+
+### onRequestTo
+
+```text
+onRequestTo([filter])
+```
+
+Parameters | Type | Description | Default
+---------- | ---- | ----------- | -----
+`filter`&#160;*(optional)* | String &#124; RegExp &#124; Object &#124; Predicate | Specifies which requests should be mocked with a response that follows in the `respond` method. See [Specifying Which Requests are Handled by the Hook](#specifying-which-requests-are-handled-by-the-hook). | All requests are mocked.
+
+```js
+var mock = RequestMock()
+    .onRequestTo('http://external-service.com/api/')
+    .respond(/*...*/)
+    .onRequestTo(/\/users\//)
+    .respond(/*...*/);
+```
+
+### respond
+
+```text
+respond(body [, statusCode] [, headers])
```

Comment:
`body` is an optional parameter too.

`.onRequestTo(...).respond()` returns an empty html page.